src/aws-cpp-sdk-core/include/aws/core/client/RetryStrategy.h

```diff
@@ -74,6 +74,15 @@ namespace Aws
              */
             virtual void RequestBookkeeping(const HttpResponseOutcome& /* httpResponseOutcome */) {}
             virtual void RequestBookkeeping(const HttpResponseOutcome& /* httpResponseOutcome */, const AWSError<CoreErrors>& /* lastError */) {}
+
+            /**
+             * Get return strategy name.
+             */
+            virtual const char* GetStrategyName() const
+            {
+                // SDK provided strategies will provide actual name
+                return "custom";
```

Comment:
why not make pure virtual?

Reply:
To maintain backward compatibility and not break build if somebody inherits from this class.